path_auth_url.go

```diff
@@ -57,7 +57,8 @@ func (b *backend) pathAuthURLWrite(ctx context.Context, req *logical.Request, da
 		RedirectURL: config.RedirectBaseURL + "/v1/monzo/callback",
 	}
 
-	url := conf.AuthCodeURL(uuid.New().String())
+	url := oauthConfig.AuthCodeURL(uuid.New().String())
+	// TODO: We probably want to persist this specific oauth2 client...
```

Comment:
Looks like we don't need to persist the entire OAuth2 client... but we definitely want to persist the uuid we just generated, so we can validate it in pathCallback